features/FEATURE_BLE/ble/gap/AdvertisingDataBuilder.h

```diff
@@ -516,56 +515,296 @@ class AdvertisingDataBuilder
     ble_error_t addData(
         adv_data_type_t advDataType,
         mbed::Span<const uint8_t> fieldData
-    )
-    {
-        return appendField(advDataType, fieldData);
+    ) {
+        uint8_t* field = findField(advDataType);
+
+        if (field) {
+            switch(advDataType.value()) {
+                /* These types are append to existing field */
+                case adv_data_type_t::INCOMPLETE_LIST_16BIT_SERVICE_IDS:
```

Comment:
It should be possible to replace these types even if it requires a memmove.

Reply:
you can replace them by calling removeData(adv_data_type_t::INCOMPLETE_LIST_16BIT_SERVICE_IDS) first

Reply:
That is not consistent with the other data type.